osf/models/node.py

```diff
@@ -1467,6 +1467,8 @@ def register_node(self, schema, auth, data, parent=None, child_ids=None, provide
         registered.tags.add(*self.all_tags.values_list('pk', flat=True))
         registered.subjects.add(*self.subjects.values_list('pk', flat=True))
         registered.affiliated_institutions.add(*self.affiliated_institutions.values_list('pk', flat=True))
+        # flatten_registration_metadata can't be called until registration has an id.
+        registered.registration_responses = registered.flatten_registration_metadata()
```

Comment:
DraftNode work passes in the DraftRegistration instead of just the `data` - there's a lot of things that need to be pulled off of the draft. Refactor this to also pass in the DraftRegistration and just pull the registration_responses off of that.